sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ModuleIdentityLifecycleSamples.cs

```diff
@@ -120,34 +119,149 @@ public async Task<ModuleIdentity> CreateModuleIdentityAsync(DeviceIdentity devic
 
-        public Task GetModuleTwinAsync()
+            try
+            {
+                Console.WriteLine($"Getting module identity with Id: '{moduleId}'\n");
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.GetIdentityAsync(deviceIdentity.DeviceId, moduleId);
+
+                ModuleIdentity moduleIdentity = response.Value;
+
+                SampleLogger.PrintSuccess($"\t- Module Id: '{moduleIdentity.ModuleId}', Device Id: '{moduleIdentity.DeviceId}', ETag: '{moduleIdentity.Etag}'");
+
+                return moduleIdentity;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to get a module identity due to:\n{ex}");
+                throw;
+            }
+        }
+        
+        /// <summary>
+        /// Update a module identity
+        /// </summary>
+        /// <param name="moduleIdentity">Module identity to be updated.</param>
+        public async Task<ModuleIdentity> UpdateModuleIdentityAsync(ModuleIdentity moduleIdentity)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("UPDATE A MODULE");
+
+            try
+            {
+                Console.WriteLine($"Updating module identity with Id: '{moduleIdentity.ModuleId}'. Setting 'ManagedBy' property to: '{Environment.UserName}'\n");
+                moduleIdentity.ManagedBy = Environment.UserName;
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.CreateOrUpdateIdentityAsync(moduleIdentity, IfMatchPrecondition.UnconditionalIfMatch);
+
+                ModuleIdentity updatedModule = response.Value;
+                
+                SampleLogger.PrintSuccess($"Successfully updated module identity with Id: '{updatedModule.ModuleId}', ManagedBy: '{updatedModule.ManagedBy}', ETag: '{updatedModule.Etag}'");
+                
+                return updatedModule;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to update a module identity due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task UpdateModuleTwinAsync()
+        /// <summary>
+        /// Get module twin.
+        /// </summary>
+        /// <param name="moduleIdentity">The module identity.</param>
+        public async Task<TwinData> GetModuleTwinAsync(ModuleIdentity moduleIdentity)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("GET A MODULE TWIN");
+
+            try
+            {
+                Console.WriteLine($"Getting module twin with Id: '{moduleIdentity.ModuleId}'\n");
+
+                Response<TwinData> response = await IoTHubServiceClient.Modules.GetTwinAsync(moduleIdentity.DeviceId, moduleIdentity.ModuleId);
+
+                SampleLogger.PrintSuccess($"\t- Module Twin Status: '{response.Value.Status}', ETag: '{response.Value.Etag}'");
```

Comment:
For I second I thought it was the `Response.Status`, then realized that it is `TwinData.Status`!